server/src/main/java/org/tctalent/server/service/db/impl/CandidateAttachmentsServiceImpl.java

```diff
@@ -143,19 +143,10 @@ public CandidateAttachment createCandidateAttachment(CreateCandidateAttachmentRe
         User user = authService.getLoggedInUser()
                 .orElseThrow(() -> new InvalidSessionException("Not logged in"));
 
-        Candidate candidate;
         String textExtract;
 
-        // Handle requests coming from the admin portal
-        if (request.getCandidateId() != null) {
-            candidate = candidateRepository.findById(request.getCandidateId())
+        Candidate candidate = candidateRepository.findById(request.getCandidateId())
```

Comment:
The request always contains the candidateId now, so we can just fetch the candidate from the request and don't need the two seperate methods.